rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JGraphLike.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+import java.util.Optional;
+
+import org.apache.commons.rdf.api.BlankNodeOrIRI;
+import org.apache.commons.rdf.api.GraphLike;
+import org.apache.commons.rdf.api.IRI;
+import org.apache.commons.rdf.api.RDFTerm;
+import org.apache.commons.rdf.api.TripleLike;
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.repository.Repository;
+
+/**
+ * Marker interface for RDF4J implementations of GraphLike.
+ * 
+ * @see RDF4JGraph
+ * 
+ */
+public interface RDF4JGraphLike<T extends TripleLike<BlankNodeOrIRI, IRI, RDFTerm>>
```

Comment:
`Triple` does inherit from `TripleLike<BlankNodeOrIRI, IRI, RDFTerm>` - that's the whole point :)

The only thing `RDFTripleLike` adds compared to `Triple` is the `asStatement()` method - which is probably useful for integration purposes.

But I think we could add a `GeneralizedTriple` interface to avoid exposure of `TripleLike` generics - but perhaps only to the jena module for now. JSONLD-Java should in theory also support that (as JSON-LD does) - at least having blank nodes for properties. 

But I think that it can be hard - at least at this point - to say what is the "common generalized triple" at the API module.